include/git2/common.h

```diff
@@ -372,6 +374,18 @@ typedef enum {
  *		> fail.  (Using the FORCE flag to checkout will still overwrite
  *		> these changes.)
  *
+ *	 opts(GIT_OPT_GET_INDEXER_MAX_OBJECTS, size_t *out)
+ *
+ *		> Get the maximum number of objects libgit2 will allow in a pack
+ *		> file when downloading a pack file from a remote. This can be
+ *		> used to limit maximum memory usage when fetching from an untrusted
+ *		> remote.
```

Comment:
Can it really? A remote is still able to send a very small amount of huge objects. So I wonder whether a size limitation would actually be more useful than a count limitation.

That being said, should this really be a global value? It could be somebody has two remotes, one of which he is trusting and another one which he is not trusting. Another scenario is packfiles which were generated locally, which have different trust levels than packfiles which are being downloaded. So I think making this an option of the indexer would be a lot saner. We'd have to provide a new way to instantiate an indexer for this, as `git_indexer_new` cannot be changed but does not have `git_indexer_opts` parameter.

Reply:
As best I can tell -- and you certainly know the code better than I -- everything *other* than the index metadata is streamed directly from the remote to disk, and so does not consume memory. And so the number of objects (which sizes the `entries` and `deltas` vectors) is the salient variable for memory (but not disk) usage.

Is there another type that has an alternate constructor that takes an `_opts` type I can look at for prior art? In addition, the `git_indexer` is constructed by the pack backend right now; What would be the idiomatic way to plumb configuration all the way through from the `remote` or `repository` to that creation?